solutions/src/test/java/ru/sberbank/school/task10/ConcurrentScalableThreadPoolTest.java

```diff
@@ -0,0 +1,166 @@
+package ru.sberbank.school.task10;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ConcurrentScalableThreadPoolTest {
+    private int max = 10;
+    private int core = 3;
+    private ThreadPool threadPool;
+
+    @Before
+    public void init() {
+        threadPool = new ConcurrentScalableThreadPool(core, max);
+    }
+
+    private Runnable newRunnable(AtomicInteger counter, CountDownLatch latch) {
+        return () -> {
+
+                System.out.println("Task: " + counter.incrementAndGet()
+                        + " - Thread: " + Thread.currentThread().getName());
+
+            latch.countDown();
+        };
+    }
+
+    private Callable<String> newCallable(AtomicInteger counter, CountDownLatch latch) {
+        return () -> {
+            String res;
+                res = "Task: " + counter.incrementAndGet() + " - Thread: " + Thread.currentThread().getName();
+
+            latch.countDown();
+            return res + " " + LocalDateTime.now().toString();
+        };
+    }
+
+    private int getThreadCount() throws InterruptedException {
+        // пауза, чтобы успели создаться/прерваться потоки до того, как я их посчитаю
+        TimeUnit.MILLISECONDS.sleep(10);
+        return Thread.getAllStackTraces().keySet().size();
+    }
+
+    @Test
+    public void normalWork() throws InterruptedException {
+        final AtomicInteger counter = new AtomicInteger(0);
+        int iterations = 100;
```

Comment:
Distance between variable &#39;iterations&#39; declaration and its first usage is 4, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a322644307>*